include/zephyr/bluetooth/bluetooth.h

```diff
@@ -388,8 +388,7 @@ uint16_t bt_get_appearance(void);
  * Automatically preserves the new appearance across reboots if
  * @kconfig{CONFIG_BT_SETTINGS} is enabled.
  *
- * This symbol is linkable if @kconfig{CONFIG_BT_DEVICE_APPEARANCE_DYNAMIC} is
- * enabled.
+ * @note Requires @kconfig{CONFIG_BT_DEVICE_APPEARANCE_DYNAMIC}.
```

Comment:
We have the `kconfig_dep` that could be used
```suggestion
 * @kconfig_dep{CONFIG_BT_DEVICE_APPEARANCE_DYNAMIC}.
```

Reply:
Ahh that's nice :)